src/commands/scale.rs

```diff
@@ -0,0 +1,423 @@
+
+#[derive(Parser, FieldNamesAsArray)]
+pub struct Args {
+    #[clap(flatten)]
+    dynamic: DynamicArgs,
+
+    /// The service to scale (defaults to linked service)
+    #[clap(long, short)]
+    service: Option<String>,
+
+    /// The environment the service is in (defaults to linked environment)
+    #[clap(long, short)]
+    environment: Option<String>,
+}
+
+pub async fn command(args: Args, _json: bool) -> Result<()> {
+    let configs = Configs::new()?;
+    let client = GQLClient::new_authorized(&configs)?;
+    let linked_project = configs.get_linked_project().await?;
+    let project = post_graphql::<queries::Project, _>(
+        &client,
+        configs.get_backboard(),
+        queries::project::Variables {
+            id: linked_project.project.clone(),
+        },
+    )
+    .await?
+    .project;
+    let environment = args
+        .environment
+        .clone()
+        .unwrap_or(linked_project.environment.clone());
+    let (existing, latest_id) = get_existing_config(&args, &linked_project, project, environment)?;
+    let new_config = convert_hashmap_into_map(
+        if args.dynamic.0.is_empty() && std::io::stdout().is_terminal() {
+            prompt_for_regions(&configs, &client, &existing).await?
+        } else if args.dynamic.0.is_empty() {
+            bail!("Please specify regions via the flags when not running in a terminal")
+        } else {
+            args.dynamic.0
+        },
+    );
+    if new_config.is_empty() {
+        println!("No changes made");
+        return Ok(());
+    }
+    let region_data = merge_config(existing, new_config);
+    handle_2fa(&configs, &client).await?;
+    update_regions_and_redeploy(configs, client, linked_project, latest_id, region_data).await?;
+
+    Ok(())
+}
+
+async fn prompt_for_regions(
+    configs: &Configs,
+    client: &reqwest::Client,
+    existing: &Value,
+) -> Result<HashMap<String, u64>> {
+    let mut updated: HashMap<String, u64> = HashMap::new();
+    let mut regions = post_graphql::<queries::Regions, _>(
+        client,
+        configs.get_backboard(),
+        queries::regions::Variables,
+    )
+    .await
+    .expect("couldn't get regions");
+    loop {
+        let get_replicas_amount = |name: String| {
+            let before = if let Some(num) = existing.get(name.clone()) {
+                num.get("numReplicas").unwrap().as_u64().unwrap() // fine to unwrap, API only returns ones that have a replica
+            } else {
+                0
+            };
+            let after = if let Some(new_value) = updated.get(&name) {
+                *new_value
+            } else {
+                before
+            };
+            (before, after)
+        };
+        regions.regions.sort_by(|a, b| {
+            get_replicas_amount(b.name.clone())
+                .1
+                .cmp(&get_replicas_amount(a.name.clone()).1)
+        });
+        let regions = regions
+            .regions
+            .iter()
+            .map(|f| {
```

Comment:
Can we filter out non-metal regions. They are available to everyone and we are in the process of migrating everyone to metal anyway.